cpp/common/src/codingstandards/cpp/rules/pointertoavirtualbaseclasscasttoapointer/PointerToAVirtualBaseClassCastToAPointer.qll

```diff
@@ -0,0 +1,34 @@
+/**
+ * Provides a library with a `problems` predicate for the following issue:
+ * A pointer to a virtual base class shall only be cast to a pointer to a derived class
+ * by means of dynamic_cast, otherwise the cast has undefined behavior.
+ */
+
+import cpp
+import codingstandards.cpp.Customizations
+import codingstandards.cpp.Exclusions
+
+abstract class PointerToAVirtualBaseClassCastToAPointerSharedQuery extends Query { }
+
+Query getQuery() { result instanceof PointerToAVirtualBaseClassCastToAPointerSharedQuery }
+
+query predicate problems(Cast cast, string message) {
+  exists(VirtualBaseClass castFrom, Class castTo |
```

Comment:
Should also be non compliant when `castFrom` is a base class of a virtual base class